Sources/JSONPatch/JSONElement.swift

```diff
@@ -435,20 +435,29 @@ extension JSONElement {
     ///
     /// - Parameters:
     ///   - operation: The operation to apply.
-    public mutating func apply(_ operation: JSONPatch.Operation) throws {
-        switch operation {
-        case let .add(path, value):
-            try add(value: value, to: path)
-        case let .remove(path):
-            try remove(at: path)
-        case let .replace(path, value):
-            try replace(value: value, to: path)
-        case let .move(from, path):
-            try move(from: from, to: path)
-        case let .copy(from, path):
-            try copy(from: from, to: path)
-        case let .test(path, value):
-            try test(value: value, at: path)
+    public mutating func apply(_ operation: JSONPatch.Operation, ignoreNonexistentValues: Bool = false) throws {
+        do {
+            switch operation {
+            case let .add(path, value):
+                try add(value: value, to: path)
+            case let .remove(path):
+                try remove(at: path)
+            case let .replace(path, value):
+                try replace(value: value, to: path)
+            case let .move(from, path):
+                try move(from: from, to: path)
+            case let .copy(from, path):
+                try copy(from: from, to: path)
+            case let .test(path, value):
+                try test(value: value, at: path)
+            }
+        } catch {
+            if let error = error as? JSONError,
+                error == .referencesNonexistentValue && ignoreNonexistentValues {
+                // Don't throw, just continue
+            } else {
+                throw error
+            }
```

Comment:
I'd liked to ask do you have a need to make JSONError conform to Equatable out with the check here? I ask because standard Swift pattern matching would have worked without the additional conformance. Since JSONError is a public type maybe you are using equality checks in your consuming app there that I'm not seeing. I make this point since for the case of patchTestFailed you need to ignore the expected and found associated values.

```suggestion
        } catch JSONError.referencesNonexistentValue where ignoreNonexistentValues {
                // Don't throw, just continue
        } catch {
                throw error
```

Reply:
The Equatable conformance is just to be able to do the `error == . referencesNonexistentValue` here. Otherwise the compiler fails with 'does not conform to Equatable'